src/clib/pio_nc.c

```diff
@@ -812,10 +820,40 @@ int PIOc_inq_var(int ncid, int varid, char *name, nc_type *xtypep, int *ndimsp,
 #ifdef _PNETCDF
         if (file->iotype == PIO_IOTYPE_PNETCDF)
         {
+            int *tmp_dimidsp = NULL;
             ierr = ncmpi_inq_varndims(file->fh, varid, &ndims);
             LOG((2, "from pnetcdf ndims = %d", ndims));
+            if(!ierr && (!dimidsp) && (file->num_unlim_dimids > 0))
+            {
+                tmp_dimidsp = (int *)malloc(ndims * sizeof(int));
+                if(!tmp_dimidsp)
+                {
+                    return pio_err(ios, file, PIO_ENOMEM, __FILE__, __LINE__);
+                }
+            }
             if (!ierr)
-                ierr = ncmpi_inq_var(file->fh, varid, name, xtypep, ndimsp, dimidsp, nattsp);
+                ierr = ncmpi_inq_var(file->fh, varid, my_name, xtypep, ndimsp, (dimidsp)?dimidsp:tmp_dimidsp, nattsp);
+            if (!ierr && name)
+            {
+                strncpy(name, my_name, PIO_MAX_NAME);
```

Comment:
Is there a reason that we should read var name to my_name first  andthen strncpy to name? The original code directly reads var name to name without the overhead of strncpy.

Reply:
There could be issues when user does not want to inquire the variable name (name is NULL)

Reply:
When name is NULL, name_present will be set to false and broadcast
We can use the logic below
if (!ierr && name) ierr = ncmpi_inq_var(file->fh, varid, name, ...)
This way we still do not need my_name (which will always be read by ncmpi_inq_var)

Reply:
We always (independent of user choice) need to query the name (to cache it)